### PR TITLE
Sample using External Secrets Operator

### PR DESCRIPTION
**Pull request by scothis**
External Secrets Operator (ESO) is a way to synthesize Kubernetes Secrets from secure data stores. There is a PR that adds provisioned service support to ExternalSecret resources.

This sample is intended to facilitate the conversation between the Service Bindings and External Secrets communities. This PR should not be merged unless and until the update PR lands.

Refs https://github.com/external-secrets/external-secrets/pull/2263

### CODE REVIEW

**Comment by scothis** [comment]
[Readable view](https://github.com/scothis/servicebinding-runtime/blob/external-secrets-sample/samples/external-secrets/README.md)

File: samples/external-secrets/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+[External Secrets][eso] is an operator to fetch content from secure stores and synthesize that state into a Kubernetes `Secret`.
+
+This sample uses a fake `SecretStore` (to avoid external infrastructure) and `ExternalSecret` to create a binding Secret. The [Spring PetClinic][petclinic] workload and MySQL database are independently bound to the `ExternalSecret`.
+
+> NOTE: Provisioned Service support within External Secrets is a proposal that is not part of any External Secrets release. This sample is to demonstrate what landing provisioned service support within ESO can enable.
+
+## Setup
+
+If not already installed, [install the ServiceBinding CRD and controller][install], and an experimental version of the [External Secrets Operator from PR #2263][pr-2263].
```
Comment (sadlerap) [inline samples/external-secrets/README.md:11]: ESO landed support with the merging of [PR 2263](https://github.com/external-secrets/external-secrets/pull/2263); should we call out that this works with [v0.8.2](https://github.com/external-secrets/external-secrets/releases/tag/v0.8.2) and later?

Comment (scothis) [reply samples/external-secrets/README.md]: I had run `git push` but didn't notice that it was stuck on an auth prompt.

**Review by sadlerap — APPROVED** [review]
LGTM

**Pull request merged**